processor/src/trace/utils.rs

```diff
@@ -237,6 +237,118 @@ impl HintCycle for u64 {
     }
 }
 
+// TRACE INFO
+// ================================================================================================
+
+/// Contains the data about lengths of the trace parts.
+///
+/// - `main_trace_len` contains the length of the main trace.
+/// - `range_table_len` contains the length of the range checker table.
+/// - `chiplet_lenghts` contains the trace lenghts of the all chiplets (hasher, bitwise, memory,
+/// kernel ROM)
+/// - `max_len` contains the trace length required to hold all execution trace steps (maximum
+/// length among range table length, main trace length and chiplets trace length).
+/// - `padded_len` contains `max_len` padded to the next power of two.
+#[allow(dead_code)]
+pub struct TraceInfo {
+    main_trace_len: usize,
+    range_table_len: usize,
+    chiplets_lenghts: ChipletsLengths,
+    max_len: usize,
+    padded_len: usize,
+}
+
+#[allow(dead_code)]
+impl TraceInfo {
+    pub fn new(
+        main_trace_len: usize,
+        range_table_len: usize,
+        chiplets_lenghts: ChipletsLengths,
+        max_len: usize,
+        padded_len: usize,
+    ) -> Self {
+        TraceInfo {
+            main_trace_len,
+            range_table_len,
+            chiplets_lenghts,
+            max_len,
+            padded_len,
+        }
+    }
+
+    /// Returns length of the main trace
+    pub fn main_trace_len(&self) -> usize {
+        self.main_trace_len
+    }
+
+    /// Returns length of the range table
+    pub fn range_table_len(&self) -> usize {
+        self.range_table_len
+    }
+
+    /// Returns [ChipletsLengths] which contains trace lengths of all chilplets.
+    pub fn chiplets_lenghts(&self) -> ChipletsLengths {
+        self.chiplets_lenghts
+    }
+
+    /// Returns maximum trace length
+    pub fn max_len(&self) -> usize {
+        self.max_len
+    }
+
+    // Returns final padded trace length
+    pub fn padded_len(&self) -> usize {
+        self.padded_len
+    }
+}
+
+/// Contains trace lengths of all chilplets: hasher, bitwise, memory and kernel ROM trace
+/// lenghts.
+#[derive(Clone, Copy)]
+pub struct ChipletsLengths {
+    hasher_chiplet_len: usize,
```

Comment:
It would be awesome if we could break out hasher chiplet length into how much of it came from hashing programs vs. verifying Merkle paths vs. everything else. But this is probably for future PRs.

Also, minor nit: we should either say `hash_chiplet_len` or `hasher_len` (I prefer the former).